indico/web/forms/fields.py

```diff
@@ -228,18 +231,75 @@ def pre_validate(self, form):
         elif self.serializable and any(p[0] == 'Group' for p in self._get_data()):
             raise ValueError(u'You cannot select groups')
 
+    def _serialize_principal(self, principal):
+        return serialize_user(principal) if isinstance(principal, User) else serialize_group(principal)
+
     def _value(self):
         if self.serializable:
-            data = retrieve_principals(self._get_data(), legacy=False)
+            principals = retrieve_principals(self._get_data(), legacy=False)
         else:
-            data = self._get_data()
-        data.sort(key=lambda x: x.name.lower())
-        return [serialize_user(x) if isinstance(x, User) else serialize_group(x) for x in data]
+            principals = self._get_data()
+        principals.sort(key=lambda x: x.full_name.lower())
+        return map(self._serialize_principal, principals)
 
     def _get_data(self):
         return sorted(self.data) if self.data else []
 
 
+class EventPersonListField(PrincipalListField):
+    """"A field that lets you select a list Indico user and EventPersons
+
+    Requires its form to have an event set.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EventPersonListField, self).__init__(*args,
+                                                   groups=False, allow_external=True, serializable=False, **kwargs)
+
+    @property
+    def event(self):
+        return self.get_form().event
+
+    def _convert_data(self, data):
+        return map(self._get_event_person, data)
+
+    def _create_event_person(self, data):
+        user = User.find_first(~User.is_deleted, User.all_emails.contains(data['email']))
+        title = next((x.value for x in UserTitle if data.get('title') == x.title), None)
+        return EventPerson(event_new=self.event, user=user, email=data['email'], _title=title,
+                           first_name=data['firstName'], last_name=data['familyName'],
+                           affiliation=data.get('affiliation'), address=data.get('address'),
+                           phone=data.get('phone'))
+
+    def _get_event_person(self, data):
+        person_type = data.get('_type')
+        if person_type is None:
+            return self._create_event_person(data)
+        if person_type == 'Avatar':
+            user = self._convert_principal(data)
+            return EventPerson.for_user(user, self.event)
+        if person_type == 'EventPerson':
+            return self.event.persons.filter_by(id=data['id']).one()
+        raise ValueError(_("Uknown person type '{}'").format(person_type))
+
+    def _serialize_principal(self, principal):
+        if not isinstance(principal, EventPerson):
+            return super(EventPersonListField, self)._serialize_principal(principal)
+        return serialize_event_person(principal)
+
+    def pre_validate(self, form):
+        return
```

Comment:
`pass` - but if you want to stop the parent's method from running put a comment to make it clear